src/lib/utils/host.ts

```diff
@@ -1,30 +1,59 @@
-const isDocker = (() => {
-  let isDockerProcess;
+const {execSync} = require("child_process");
+const {anchoredValue, DAPP_PATH} = require("../core/env");
+const {hostname} = require("os");
+
+const dappPath = anchoredValue(DAPP_PATH);
 
-  const hostname = require("os").hostname();
-  const pattern = new RegExp(
-    "[0-9]+\:[a-z_-]+\:\/docker\/" + hostname + "[0-9a-z]+", "i",
-  );
+function subdir(pdir_, dir_) {
+  let pdir = path.resolve(path.normalize(pdir_)) + (path.sep || "/");
+  const dir = path.resolve(pdir, path.normalize(dir_));
+  if (pdir === "//") { pdir = "/"; }
+  if (pdir === dir) { return false; }
+  return dir.slice(0, pdir.length) === pdir;
+}
+
+const isDocker = (() => {
+  // assumption: an Embark container is always a Linux Docker container, though
+  // the Docker host may be Linux, macOS, or Windows
+  if (process.platform !== "linux") { return false; }
+  try {
+    return (
+      new RegExp(`[0-9]+\:[a-z_-]+\:\/docker\/${hostname()}[0-9a-z]+`, "i")
+    ).test(
+      execSync(
+        "cat /proc/self/cgroup",
+        {stdio: ["ignore", "pipe", "ignore"]},
```

Comment:
For the `execSync` calls in this module, all we care about is `stdout`, so it seems more correct to `'ignore'` `stdin` and `stderr`. It would work with the default, though: `'pipe'`, which is equivalent to `['pipe', 'pipe', 'pipe']`.

Reply:
Yea, that makes sense!